code/modules/vehicles/arc/arc.dm

```diff
@@ -0,0 +1,263 @@
+
+/obj/vehicle/multitile/arc/crew_mousedown(datum/source, atom/object, turf/location, control, params)
+	return
+
+
+/obj/vehicle/multitile/arc/get_examine_text(mob/user)
+	. = ..()
+	if(!isxeno(user))
+		return
+
+	if(health > 0)
+		. += SPAN_XENO("[src] can be crawled under once destroyed.")
+	else
+		. += SPAN_XENO("[src] can be crawled under by <b>dragging our sprite</b> to it.")
+
+/obj/vehicle/multitile/arc/proc/on_antenna_toggle(datum/source)
+	SIGNAL_HANDLER
+
+	if(antenna_deployed)
+		START_PROCESSING(SSslowobj, src)
+
+	else
+		STOP_PROCESSING(SSslowobj, src)
+
+/obj/vehicle/multitile/arc/process()
+	var/turf/arc_turf = get_turf(src)
+	if((health <= 0) || !visible_in_tacmap || !is_ground_level(arc_turf.z))
+		return
+
+	var/obj/item/hardpoint/support/arc_antenna/antenna = locate() in hardpoints
+	if(!antenna || (antenna.health <= 0))
+		for(var/datum/weakref/xeno as anything in minimap_added)
+			SSminimaps.remove_marker(xeno.resolve())
+			minimap_added.Remove(xeno)
+		return
+
+	for(var/mob/living/carbon/xenomorph/current_xeno as anything in GLOB.living_xeno_list)
+		var/turf/xeno_turf = get_turf(current_xeno)
+		if(!is_ground_level(xeno_turf.z))
+			continue
+
+		if(get_dist(src, current_xeno) <= sensor_radius)
+			if(WEAKREF(current_xeno) in minimap_added)
+				continue
+
+			SSminimaps.remove_marker(current_xeno)
+			current_xeno.add_minimap_marker(MINIMAP_FLAG_USCM|MINIMAP_FLAG_XENO)
+			minimap_added += WEAKREF(current_xeno)
+		else
+			if(WEAKREF(current_xeno) in minimap_added)
+				SSminimaps.remove_marker(current_xeno)
+				current_xeno.add_minimap_marker()
+				minimap_added -= WEAKREF(current_xeno)
+
+/obj/vehicle/multitile/arc/relaymove(mob/user, direction)
+	if(antenna_deployed)
+		return FALSE
+
+	return ..()
+
+/obj/vehicle/multitile/arc/load_role_reserved_slots()
+	var/datum/role_reserved_slots/RRS = new
+	RRS.category_name = "CIC Officer"
+	RRS.roles = list(JOB_SO, JOB_SEA, JOB_XO, JOB_CO, JOB_GENERAL)
+	RRS.total = 2
+	role_reserved_slots += RRS
+
+	RRS = new
+	RRS.category_name = "Intelligence Officer"
+	RRS.roles = list(JOB_INTEL)
+	RRS.total = 1
+	role_reserved_slots += RRS
+
+/obj/vehicle/multitile/arc/set_seated_mob(seat, mob/living/M)
+	. = ..()
+	if(!.)
+		return
+
+	give_action(M, /datum/action/human_action/toggle_arc_antenna)
+
+/obj/vehicle/multitile/arc/add_seated_verbs(mob/living/M, seat)
+	if(!M.client)
+		return
+	add_verb(M.client, list(
+		/obj/vehicle/multitile/proc/get_status_info,
+		//obj/vehicle/multitile/proc/open_controls_guide,
+		/obj/vehicle/multitile/proc/toggle_door_lock,
+		/obj/vehicle/multitile/proc/activate_horn,
+		/obj/vehicle/multitile/proc/name_vehicle,
+		/obj/vehicle/multitile/arc/proc/toggle_antenna,
+	))
+
+/obj/vehicle/multitile/arc/remove_seated_verbs(mob/living/M, seat)
+	if(!M.client)
+		return
+	remove_verb(M.client, list(
+		/obj/vehicle/multitile/proc/get_status_info,
+		//obj/vehicle/multitile/proc/open_controls_guide,
```

Comment:
thanks for reminding me to fix this. Proc is set to a unique one (instead of inheriting) due to adding/removing verbs not always respecting inheritance